### PR TITLE
Add integration test workflow

### PR DESCRIPTION
**Pull request by astaticvoid**
*Issue #, if available:*
#69 

*Description of changes:*
Run integration test for push and pull requests. Uses repo environment secrets to setup AWS credentials.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by astaticvoid** [comment]
It appears the integration test is not useful for pull requests since the environment secrets will not be available if PR is from forked repo, even if the deployment is approved by us. 
We can use it on push, which gives us confidence before releasing at least. Also provision to run locally gives us some mostly automated assurances before creating the PR.

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  integration-test:
+    name: Run Integration- Test
+    runs-on: ubuntu-latest
+    environment: Integration Test
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (bendu) [inline .github/workflows/integration-test.yaml:19]: We could use the github actions OIDC provider instead, so we don't have to remember to rotate the creds.

See https://github.com/aws-actions/configure-aws-credentials#assuming-a-role and https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services

File: .github/workflows/integration-test.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Comment (bendu) [inline .github/workflows/integration-test.yaml:21]: Also is kind able to pick up on these creds? I feel we may have to do some extra steps.

**Comment by runakash** [comment]
> It appears the integration test is not useful for pull requests since the environment secrets will not be available if PR is from forked repo, even if the deployment is approved by us.

How about we run this periodically just for the `main` branch?

**Comment by bendu** [comment]
> > It appears the integration test is not useful for pull requests since the environment secrets will not be available if PR is from forked repo, even if the deployment is approved by us.
> 
> How about we run this periodically just for the `main` branch?

I'm good with either periodic or just on push.

File: .github/workflows/integration-test.yaml (hunk 2 above)
Comment (astaticvoid) [reply .github/workflows/integration-test.yaml:21]: It worked on the [private branch](https://github.com/astaticvoid/aws-cloud-map-mcs-controller-for-k8s/runs/4162413311?check_suite_focus=true)

Comment (bendu) [reply .github/workflows/integration-test.yaml:21]: great!

File: .github/workflows/integration-test.yaml (hunk 1 above)
Comment (astaticvoid) [reply .github/workflows/integration-test.yaml:19]: Thanks, we should do that.
It was my original intent, but I hit some policy snags setting up our testing account. Let me take another crack at it for the next revision.

**Review by bendu — APPROVED** [review]

**Pull request merged**